src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaService.java

```diff
@@ -27,4 +28,6 @@
     List<SagaLog> getLogsByTransaction(String txId);
 
     Page<SagaTransaction> getAllTransaction(Pageable pageable);
+
+    Optional<SagaTransaction> findSagaTransactionById(String id);
```

Comment:
@dtsarenko suggest to rename `findSagaTransactionById` to `findTransactionById` here and in the service